### ISSUE TITLE
Is curriculum-learning for parameter is effective?

### ISSUE THREAD

**Issue by dora-gt** [Unity-Technologies/ml-agents#133]
Hi,

I have some questions.
I really appreciate your help. Could you give me advice?

1. Is curriculum-learning for parameter is effective?
For example... if a model has nearly 200 parameters, and most parameters are meaningless for easier lesson (an easy part of curriculum), then does passing 0 value to the meaningless parameters not to confuse the model accelerate learning? I'm now facing situation that a model can't learn from so many parameters.

2. Is there any effective way to train with so many (nearly 200) parameters?
Like the above, I have problem with train with so many parameters. Do you have any knowledge to train a model like it? Need more hidden-units? lower learning rate? or something else?

Thank you so much for reading.

**Comment by surferau** [comment]
200 parameters is not too much. For example when working with images each pixel counts as an input. 100x100 image = 10000 inputs.

The problem is that there many different ways to implement nn model like what layers too choose (https://www.tensorflow.org/api_docs/python/tf/layers/dense) while mlagents PPO tries general approach with densely connected layers, does not mean that it will work in every case.

**Comment by dora-gt** [comment]
@surferau thank you for the reply.

I'm trying to understand PPO and think whether I can modify the prepared model, not losing essence of PPO.

Thank you anyway.

**Comment by awjuliani** [comment]
Hi @dora-gt,

Keeping space for parameters that are currently irrelevant shouldn't be an issue for the model, as long as they are logged as zeros.
